code/core/src/cli/helpers.ts

```diff
@@ -197,42 +199,44 @@ export async function copyTemplateFiles({
     const assetsTS38 = join(assetsDir, languageFolderMapping[SupportedLanguage.TYPESCRIPT_3_8]);
 
     // Ideally use the assets that match the language & version.
-    if (await pathExists(assetsLanguage)) {
+    if (existsSync(assetsLanguage)) {
```

Comment:
Should we keep this async maybe? Assuming there's an async fs method for checking existence

Reply:
There's no Promise-based alternative, and the callback one is marked as deprecated: https://nodejs.org/docs/latest-v18.x/api/fs.html#fsexistspath-callback. This is why I opted for `existsSync` as awaiting the promise is effectively blocking too, so I think it's rather similar.

The documentation suggests using `fs.stat()` or `fs.access()` for checking the existence asynchronously, but I found that kinda clunky: you can see an example of how that looks in `configure.ts` (https://github.com/storybookjs/storybook/pull/29126/files#diff-27fc97f1960770b2c6cfeeb0f0c1f852df6d4d3a190a3ea293c7c864e97dddab).

If that approach is preferred by the maintainers nonetheless, I'm not opposed to changing it! 😄

Reply:
Also, the documentation raises a valid concern related to existence checks:

"Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile(), or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist."

I wonder if I should rewrite such usages into the suggested approach to avoid race conditions. Thoughts?

Reply:
I'm not too concerned with that, let's keep it as close as possible to the existing flow. in practice these files here aren't being modified simultaneously by other processes.